internal/engine/compiler/engine.go

```diff
@@ -705,6 +705,34 @@ func (ce *callEngine) Definition() api.FunctionDefinition {
 
 // Call implements the same method as documented on wasm.ModuleEngine.
 func (ce *callEngine) Call(ctx context.Context, params ...uint64) (results []uint64, err error) {
+	ft := ce.initialFn.funcType
+	if n := ft.ParamNumInUint64; n != len(params) {
+		return nil, fmt.Errorf("expected %d params, but passed %d", n, len(params))
+	}
+	return ce.call(ctx, params, nil)
+}
+
+// CallWithStack implements the same method as documented on wasm.ModuleEngine.
+func (ce *callEngine) CallWithStack(ctx context.Context, stack []uint64) error {
+	var params, results []uint64
+
+	ft := ce.initialFn.funcType
+	if n := ft.ParamNumInUint64; n > len(stack) {
```

Comment:
I extracted the logic here and in interpreter into a different PR to test it https://github.com/tetratelabs/wazero/pull/1414

Reply:
Thanks for the refactor, saves me writing a bunch of test code.